docs/styleguides/python.md

```diff
@@ -0,0 +1,173 @@
+# DCOS Python Style Guide
+###### Version 0.1
+###### April-6-2017
+
+For Python source code within DCOS, we prefer to keep Google Python Style Guide Revision 2.59 as the base and **extend** it with
+some changes that will make DCOS codebase look modern, consistent and easy to follow.
+
+##### Rationale
+
+With any style guide, the primary requirement is to adopt one and follow it consistently. With our choices below, we can use tools available in the ecosystem to our advantage, enforce consistency in the codebase, creating and publishing automatically generated documentation for our code.
+
+##### Versioned document
+
+Styleguide _conventions_ are not set in stone. We can improve upon this or remove inconvenient styles by creating a PR, discussing the merits or demerits to arrive at a consensus.
+
+#### Base Style Guide
+
+* [Google Python Style Guide Revision 2.59](https://google.github.io/styleguide/pyguide.html)
+
+#### Summary of Style Requirements DCOS which extend Base Style Guide and are specific to DCOS.
+
+* Import Requirements
+    * Adopt the base style guide preferences.
+
+        ```
+        Use import x for importing packages and modules.
+        Use from x import y where x is the package prefix and y is the module name with no prefix.
+        Use from x import y as z if two modules named y are to be imported or if y is an inconveniently long name.
+        ```
+    * Use `__init__.py` only to represent python packages.
+        * Avoid importing of objects in `__init__.py` for readability.
+* Import formatting
+
+    * The types of imports should be separated by single line and order is should be.
+        * standard library imports.
+        * third-party imports.
+        * application specific imports.
+
+    * The `from` style import and `import` statements should be separated by single line too.
+    * Each group of these imports should be sorted.
+
+For Example
+
+```
+import json
+import uuid
+
+from subprocess import CalledProcessError
+from typing import Callable, List
+
+import retrying
+
+from retrying import retry, RetryError
+from teamcity.messages import TeamcityServiceMessages
+
+import test_util.aws
+import test_util.cluster
+
+from pkgpanda.util import logger, write_string
+```
+
+* [Packages](https://google.github.io/styleguide/pyguide.html?showone=Packages#Packages)
+    * Import each module using the full pathname location of the module.
+    * No Relative Imports.
+
+* Maximum line length is 120 characters.
+    * Following are the exceptions to this rule. These types can be of arbitrary length.
+        * Long import statements.
+        * URLs in comments.
+        * Any sequence of chars which would better be in a single line.
+
+* Indentation
+
+    * Use 4space indentation.
+    * For long list of arguments, use double-indented hanging-indentation.
+ For e.g.
+    * The hanging indentation should be followed by a single blank line.
```

Comment:
You propose a simple new rule to encompass a preexisting complicated rule. Usually this would bother me, but come to think of it, it took me forever to learn how to make the style editor happy with hanging indents, so I can go along with this :+1: